packages/date-picker/src/vaadin-date-picker-helper.js

```diff
@@ -103,3 +103,28 @@ export function extractDateParts(date) {
     year: date.getFullYear(),
   };
 }
+
+/**
+ * Calculate the year of the date based on the provided reference date
+ * @param {!Date} referenceDate The date to act as basis in the calculation
+ * @param {!Number} year Should be in the range of [0, 99]
+ * @param {Number} month
+ * @param {Number} date
+ * @return {!Number} Adjusted year value
+ */
+export function calculateYearBasedOnReferenceDate(referenceDate, year, month, date) {
```

Comment:
The function could use a shorter name, how about `getAdjustedYear()`?

Reply:
Addressed in the latest commit.